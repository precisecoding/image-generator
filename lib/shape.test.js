const { Triangle, Square, Circle } = require('./shapes');

describe('Circle', () => {
    it('Should generate a circle w/ the rest of user inputs', () => {
        const circle = new Circle('blue');
        expect(circle.render('abc', 'black')).toMatchSnapshot()
    })
    // .toMatchSnapshot() works for svg/html/README/etc... to generate output with values provided above rather than me typing it manually like in the miniproject examples.
})

describe('square', () => {
    it('Should generate a square w/ the rest of user inputs', () => {
        const square = new Square('purple');
        expect(square.render('def', 'red')).toMatchSnapshot()
    })
})

describe('triangle', () => {
    it('Should generate a triangle w/ the rest of user inputs', () => {
        const triangle = new Triangle('pink');
        expect(triangle.render('ghi', 'blue')).toMatchSnapshot()
    })
})
