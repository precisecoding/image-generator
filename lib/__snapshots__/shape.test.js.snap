// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Circle Should generate a circle w/ the rest of user inputs 1`] = `
"<svg version=\\"1.1\\" width=\\"300\\" height=\\"200\\" xmlns=\\"http://www.w3.org/2000/svg\\">
        <circle cx=\\"150\\" cy=\\"100\\" r=\\"80\\" fill=\\"blue\\"/>
        <text x=\\"150\\" y=\\"125\\" font-size=\\"60\\" text-anchor=\\"middle\\" fill=\\"black\\">abc</text></svg>"
`;

exports[`square Should generate a square w/ the rest of user inputs 1`] = `
"<svg version=\\"1.1\\" width=\\"300\\" height=\\"200\\" xmlns=\\"http://www.w3.org/2000/svg\\">
        <rect width=\\"300\\" height=\\"100\\" style=\\"fill:purple;\\"/>
        <text x=\\"150\\" y=\\"65\\" font-size=\\"60\\" text-anchor=\\"middle\\" fill=\\"red\\">def</text></svg>"
`;

exports[`triangle Should generate a triangle w/ the rest of user inputs 1`] = `
"<svg version=\\"1.1\\" width=\\"300\\" height=\\"200\\" xmlns=\\"http://www.w3.org/2000/svg\\">
        <polygon points=\\"100,10 40,198 190,78 10,78 160,198\\"
        style=\\"fill:pink;\\"/>
        <text x=\\"105\\" y=\\"120\\" font-size=\\"40\\" text-anchor=\\"middle\\" fill=\\"blue\\">ghi</text></svg>"
`;
